sdk/keyvault/keyvault/src/secrets/index.ts

```diff
@@ -0,0 +1,470 @@
+    }
+    return userAgentInfo.join(" ");
+  }
+
+  // TODO: do we want Aborter as well?
+
+  /**
+   * The SET operation adds a secret to the Azure Key Vault. If the named secret already exists,
+   * Azure Key Vault creates a new version of that secret. This operation requires the secrets/set
+   * permission.
+   * @summary Adds a secret in a specified key vault.
+   * @param secretName The name of the secret.
+   * @param value The value of the secret.
+   * @param [options] The optional parameters
+   * @returns Promise<Secret>
+   */
+  public async setSecret(
+    secretName: string,
+    value: string,
+    options?: SetSecretOptions
+  ) {
+    if (options) {
+      let unflattenedAttributes = { enabled: options.enabled, notBefore: options.notBefore, expires: options.expires };
+      let unflattenedOptions = { ...options, ...(options.requestOptions ? options.requestOptions : {}), secretAttributes: unflattenedAttributes };
+      delete unflattenedOptions.enabled;
+      delete unflattenedOptions.notBefore;
+      delete unflattenedOptions.expires;
+      delete unflattenedOptions.requestOptions;
+
+      const response = await this.client.setSecret(this.vaultBaseUrl, secretName, value, unflattenedOptions);
+      return this.getSecretFromSecretBundle(response);
+    } else {
+      const response = await this.client.setSecret(this.vaultBaseUrl, secretName, value, options);
+      return this.getSecretFromSecretBundle(response);
+    }
+  }
+
+  /**
+   * The DELETE operation applies to any secret stored in Azure Key Vault. DELETE cannot be applied
+   * to an individual version of a secret. This operation requires the secrets/delete permission.
+   * @summary Deletes a secret from a specified key vault.
+   * @param vaultBaseUrl The vault name, for example https://myvault.vault.azure.net.
+   * @param secretName The name of the secret.
+   * @param [options] The optional parameters
+   * @returns Promise<DeletedSecret>
+   */
+  public async deleteSecret(
+    secretName: string,
+    options?: RequestOptionsBase
+  ): Promise<DeletedSecret> {
+    const response = await this.client.deleteSecret(this.vaultBaseUrl, secretName, options);
+    return this.getSecretFromSecretBundle(response);
+  }
+
+  /**
+   * The UPDATE operation changes specified attributes of an existing stored secret. Attributes that
+   * are not specified in the request are left unchanged. The value of a secret itself cannot be
+   * changed. This operation requires the secrets/set permission.
+   * @summary Updates the attributes associated with a specified secret in a given key vault.
+   * @param secretName The name of the secret.
+   * @param secretVersion The version of the secret.
+   * @param [options] The optional parameters
+   * @returns Promise<Secret>
+   */
+  public async updateSecretAttributes(
+    secretName: string,
+    secretVersion: string,
+    options?: UpdateSecretOptions
+  ): Promise<Secret> {
```

Comment:
>Promise<Secret> [](start = 5, length = 15)

I'm not sure but I believe this will not return the secret value.  If this is the case the return type should be  Promise<SecretAttributes>

Reply:
In coordination with @jonathandturner, let's address this through this issue: https://github.com/Azure/azure-sdk-for-js/issues/3030